### DISCUSSION TITLE
add a parameter (EditEvent?) on FluentTextField (and other components) to manage onchange/oninput binding

### DISCUSSION THREAD

**Discussion by sebestyn168** [microsoft/fluentui-blazor#297] [Ideas]
Would you consider adding a parameter EditEvent on FluentTextField to take into account the binding event (onchange/oninput) and avoid subclassing the components?

Basically adding the parameter in the class and Rendering the component in code is working for me

```
    [Parameter] public EditEvent EditEvent { get; set; }

    protected override void BuildRenderTree(RenderTreeBuilder builder)
    {
        builder.OpenElement(0, "fluent-text-field");
        builder.AddMultipleAttributes(1, AdditionalAttributes);

        builder.AddAttribute(2, "type", TextFieldType.ToAttributeValue());
        builder.AddAttribute(3, "value", Value);
        builder.AddAttribute(4, "current-value", BindConverter.FormatValue(CurrentValue));

        builder.AddAttribute(5, EditEvent.ToString().ToLower(), EventCallback.Factory.CreateBinder<string?>(this, __value => CurrentValueAsString = __value, CurrentValueAsString));

        builder.AddAttribute(6, "id", Id);
        builder.AddAttribute(7, "name", Name);
        builder.AddAttributeIfNotNullOrEmpty(8, "class", Class);
        builder.AddAttributeIfNotNullOrEmpty(9, "style", Style);

        builder.AddAttribute(10, "readonly", Readonly);
        builder.AddAttribute(11, "disabled", Disabled);
        builder.AddAttribute(12, "autofocus", Autofocus);
        builder.AddAttribute(13, "placeholder", Placeholder);
        builder.AddAttribute(14, "appearance", Appearance.ToAttributeValue());
        builder.AddAttribute(15, "list", DataList);
        builder.AddAttribute(16, "spellcheck", Spellcheck);

        builder.AddAttribute(17, "required", Required);
        builder.AddAttribute(18, "minlength", Minlength);
        builder.AddAttribute(19, "maxlength", Maxlength);
        builder.AddAttribute(20, "size", Size);
        builder.AddAttribute(21, "pattern", Pattern);

        builder.AddElementReferenceCapture(22, @ref => Element = @ref);
        builder.AddContent(23, ChildContent);
        builder.CloseElement();
    }
```

**Comment by vnbaaij** [comment]
What type would `EditEvent` be? It's not a built-in thing, is it?

**Comment by sebestyn168** [comment]
I added an enum with OnChange, OnInput values so ToLower matches the expected values (not sure if it's a good idea)

Also I used the extension AddAttributeIfNotNullOrEmpty from Microsoft.Fast.Components.FluentUI.Extensions (Net7)

    public static class FluentRenderTreeBuilderExtensions
    {
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static void AddAttributeIfNotNullOrEmpty(this RenderTreeBuilder builder, int sequence, string name, string? value)
        {
            if (!string.IsNullOrEmpty(value))
            {
                builder.AddAttribute(sequence, name, value);
            }
        }
    }

**Comment by vnbaaij** [comment]
I'm moving this to the discussion area to start a discussion (duh) on this. Would like to know how others think about this,

**Reply by sebestyn168** [reply]
Parameters Immediate (+ImmediateDelay) have been added in v3.1
